x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/components/datasource_delete_provider.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useMemo, useRef, useState } from 'react';
+import { EuiCallOut, EuiConfirmModal, EuiOverlayMask, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useCore, sendRequest, sendDeleteDatasource } from '../../../hooks';
+import { AGENT_API_ROUTES } from '../../../../../../common/constants';
+import { AgentConfig } from '../../../../../../common/types/models';
+
+interface Props {
+  agentConfig: AgentConfig;
+  children: (deleteDatasourcePrompt: DeleteAgentConfigDatasourcePrompt) => React.ReactElement;
+}
+
+export type DeleteAgentConfigDatasourcePrompt = (
+  datasourcesToDelete: string[],
+  onSuccess?: OnSuccessCallback
+) => void;
+
+type OnSuccessCallback = (datasourcesDeleted: string[]) => void;
+
+export const DatasourceDeleteProvider: React.FunctionComponent<Props> = ({
+  agentConfig,
+  children,
+}) => {
+  const { notifications } = useCore();
+  const [datasources, setDatasources] = useState<string[]>([]);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [isLoadingAgentsCount, setIsLoadingAgentsCount] = useState<boolean>(false);
+  const [agentsCount, setAgentsCount] = useState<number>(0);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const onSuccessCallback = useRef<OnSuccessCallback | null>(null);
+
+  const fetchAgentsCount = useMemo(
+    () => async () => {
+      if (isLoadingAgentsCount) {
+        return;
+      }
+      setIsLoadingAgentsCount(true);
+      const { data } = await sendRequest<{ total: number }>({
+        path: AGENT_API_ROUTES.LIST_PATTERN,
+        method: 'get',
+        query: {
+          page: 1,
+          perPage: 1,
+          kuery: `agents.config_id : ${agentConfig.id}`,
+        },
+      });
+      setAgentsCount(data?.total || 0);
+      setIsLoadingAgentsCount(false);
+    },
+    [agentConfig.id, isLoadingAgentsCount]
+  );
+
+  const deleteDatasourcesPrompt = useMemo(
+    (): DeleteAgentConfigDatasourcePrompt => (datasourcesToDelete, onSuccess = () => undefined) => {
+      if (!Array.isArray(datasourcesToDelete) || datasourcesToDelete.length === 0) {
+        throw new Error('No datasources specified for deletion');
+      }
+      setIsModalOpen(true);
+      setDatasources(datasourcesToDelete);
+      fetchAgentsCount();
```

Comment:
Checking # of agents would only be needed if fleet is enabled

Reply:
Ahh - yes. Will add check for it.